### DISCUSSION TITLE
Change query just for view page

### DISCUSSION THREAD

**Discussion by joaovdiasb** [filamentphp/filament#3876] [Help] [answered]
I changed the default query from the listing and also changed from the view, how change just the query from ListPage and not affect ViewPage?

**Comment by danharrin** [comment, marked as answer]
Instead of extending `getEloquentQuery()` on the resource class, extend `getTableQuery()` on the list page class

**Reply by joaovdiasb** [reply]
![CleanShot 2022-09-05 at 16 01 13](https://user-images.githubusercontent.com/51106311/188503067-ed1d7891-761b-4d97-804e-b36cd0ace1d6.png)
it's already on the ListPage using getTableQuery

**Reply by danharrin** [reply]
I dont understand what you want help with then?

**Reply by joaovdiasb** [reply]
> I dont understand what you want help with then?


when change this listing query, break when you view some data

**Reply by joaovdiasb** [reply]
<img width="907" alt="CleanShot 2022-09-06 at 11 03 34@2x" src="https://user-images.githubusercontent.com/51106311/188655714-1b69fd74-3a63-4259-834e-84aac18e7cab.png">

**Reply by danharrin** [reply]
Some part of your query must be breaking Eloquent then?

**Reply by joaovdiasb** [reply]
> Some part of your query must be breaking Eloquent then?

yes, the question is if I can change the query just for the listing, and not affect the view

**Reply by danharrin** [reply]
The view action uses the query to get data, so I don't think so. You'll need to work out how to avoid breaking the Eloquent query in the first placr

**Reply by joaovdiasb** [reply]
Thanks, I will work on it then

**Comment by mikestmike** [comment]
I had an integrity constraint error on id of the resource model
This fixed the issue.
